File: lecture_003(Pattern)/l_007_pattern7.cpp
#include<iostream>
using namespace std;

void patterndiamondnumber2(int n){
    int nst=1;
    int nsp=n/2;
    int num=0;  
    for (int r=1; r<=n; r++) { 
        for (int csp=1; csp<=nsp; csp++) {
           cout<<" ";
        }
        for (int cst=1; cst<=nst; cst++) {
            num++;
            cout<<num;
        }
        if (r<=n/2) {
        nsp--;
        nst+=2; 
        }
        else {
            nst-=2;
            nsp++;
        }
        cout<<endl;
        }
}

int main(int args, char**argv) {
    int n;
    cin>>n;
    patterndiamondnumber2(n);
    return 0;
}

File: lecture_003(Pattern)/l_007_pattern7.java
import java.util.Scanner;
public class l_007_pattern7 {
    public static Scanner scn = new Scanner(System.in);
    public static void main(String[] args) {
        int n = scn.nextInt();
        patternDiamondNumber2(n);
    }
    public static void patternDiamondNumber2(int n) {
        int nst=1;
        int nsp=n/2;
        int num = 0;
        for (int r=1; r<=n; r++) {
            for (int csp=1; csp<=nsp; csp++) {
                System.out.print(" ");
            }
            for (int cst=1; cst<=nst; cst++) {
                num++;
                System.out.print(num);
            }
            if(r <= n/2) {
                nst+=2;
                nsp--;
            }
            else {
                nst-=2;
                nsp++;
            }
            System.out.println();
        }
    }
}